TrackletAlgorithm/MatchCalculator.h

```diff
@@ -334,26 +335,21 @@ void MatchCalculator(BXType bx,
   CandidateMatch::CMProjIndex id;
   CandidateMatch::CMProjIndex id_next;
 
-  ap_uint<kNBits_MemAddr> ncm1 = 0; 
-  ap_uint<kNBits_MemAddr> ncm2 = 0;
-  ap_uint<kNBits_MemAddr> ncm3 = 0;
-  ap_uint<kNBits_MemAddr> ncm4 = 0;
-  ap_uint<kNBits_MemAddr> ncm5 = 0;
-  ap_uint<kNBits_MemAddr> ncm6 = 0;
-  ap_uint<kNBits_MemAddr> ncm7 = 0;
-  ap_uint<kNBits_MemAddr> ncm8 = 0;
+  ap_uint<kNBits_MemAddr> ncm[totalMatchCopies];
+#pragma HLS array_partition variable=ncm complete dim=0
+  ncm_loop: for(int i = 0; i < totalMatchCopies; ++i) { // priority encoder ALWAYS expects 8
+#pragma HLS unroll
+    ncm[i] = ap_uint<kNBits_MemAddr>(0);
+  }
   ap_uint<kNBits_MemAddr+1> total  = 0;
-  ap_uint<kNBits_MemAddr> ncm = 0;
 
   // Initialize read addresses for candidate matches
-  ap_uint<kNBits_MemAddr> addr1 = 0;  
-  ap_uint<kNBits_MemAddr> addr2 = 0;  
-  ap_uint<kNBits_MemAddr> addr3 = 0;  
-  ap_uint<kNBits_MemAddr> addr4 = 0;  
-  ap_uint<kNBits_MemAddr> addr5 = 0;  
-  ap_uint<kNBits_MemAddr> addr6 = 0;  
-  ap_uint<kNBits_MemAddr> addr7 = 0;  
-  ap_uint<kNBits_MemAddr> addr8 = 0; 
+  ap_uint<kNBits_MemAddr> addr[totalMatchCopies];
+#pragma HLS array_partition variable=addr complete dim=0
+  addr_loop: for(int i = 0; i < totalMatchCopies; ++i) { // priority encoder ALWAYS expects 8
+#pragma HLS unroll
+    addr[i] = 0;
+  }
 
   // Read signals for the input candidate matches
   bool read1 = false;
```

Comment:
@bryates It looks like `read1`, `read2`, …, and `read1_next`, `read2_next`, … are no longer needed. Can they be removed?

Reply:
Good point. I've removed them.